enhancements/installer/storage-class-encrypted.md

```diff
@@ -0,0 +1,350 @@
+implement this feature when that could be handled day 2 by a user. The complexity of this task is quite small and seems
+reasonable, so this drawback does not seem like a reason to block this enhancement.
+
+## Design Details
+
+### Open Questions 
+
+1. Should these API extensions go in both the spec & status as proposed?
+2. When should the installer write the keys to the cluster infrastructure object; particularly which fields in the
+install config correlate to "set default storage class"? Is the `defaultMachinePool`, as proposed the best fit? What
+about if control-plane/compute nodes only are set? What about a new field in the install config
+([see alternatives below](#alternatives)).
+3. Should e2e-tests be required for this feature?
+
+### Test Plan
+
+If desired, presubmit and/or periodic jobs could follow this pattern:
+1. In pre-steps, create or use existing keys to populate install config.
+2. Create cluster
+3. Add a step to ensure default storage class uses specified keys.
+
+### Graduation Criteria
+
+#### GA
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+### Upgrade / Downgrade Strategy
+
+After upgrade, users should be able to utilize the new API fields to have the cluster-storage-operator or CSI driver
+set the default storage class. Presumably, the operator/driver would do nothing if the fields are unset; so the
+upgrade should be safe.
+
+### Version Skew Strategy
+
+Version skew should not be an issue for this enhancement.
+
+### Operational Aspects of API Extensions
+
+- Sets default storage class when populated on cluster infrastructure object
+
+#### Failure Modes
+
+- If bad data is entered into the cluster infrastructure object for this use case, the operator/driver should fail
+to set the default storage class. 
+- The installer and storage teams should be involved 
+
+#### Support Procedures
+
+- The cluster-storage-operator/driver should validate the existence/availability of the keys
+- Failure of validation should appear in logs and stop further action
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Alternatives
+
+Instead of using the existing fields in the machinepools of the install config, it would be possible to define a new
+top-level field to define a default storage class. I have decided against this idea, because it has a broader scope
+and this proposal does not preclude the future introduction of such a field. 
+
+### API Alternatives
+
+The initial version of this enhancement provided a more structured implementation of the Azure & GCP API fields, which are included below.
+These have been updated to a single string for two reasons: 1. To be consistent with AWS 2. To be in-line with upstream. 
```

Comment:
I like the sound of 1, but I don't like the sound of 2, we can convert between the two easily right? We know the format internally.

Seems like the original style is actually better UX wise for a user, I'd be tempted to switch back to it